### PR TITLE
FEAT: Add calendar, Fixed Lint errors

### PR DESCRIPTION
**Pull request by vanshavenger**
In this i have added a calendar component for the Cohort students and admin can access and update events, so that students can just come and check on which day which topic in class happened and find its video link and Notes attached. And Students can see upcoming agenda at the single place that is updated by the admin of the next entire 30 days from the current date.

Attaching images of the added page, And proof of no lint errors.

Screenshot 2024-08-16 at 3 24 57 PM
Screenshot 2024-08-16 at 3 24 45 PM
Screenshot 2024-08-16 at 3 24 38 PM
Screenshot 2024-08-16 at 3 24 29 PM
Screenshot 2024-08-16 at 3 26 25 PM

### Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I assure there is no similar/duplicate pull request regarding same issue

### CODE REVIEW

**Comment by raashid16** [comment]
Nice addition

**Comment by hkirat** [comment]
@devsargam @siinghd

**Comment by hkirat** [comment]
If it gets merged, /bounty $100

**Comment by devsargam** [comment]
Reviewing

**Review by devsargam — CHANGES_REQUESTED** [review]
Apart from all those. Works perfectly. 😄

**Comment by vanshavenger** [comment]
@devsargam fixed all the changes you asked for, kindly check. The calendar script was there as that can be used for production use case also, as it wil fill all blocks of friday, sat,sunday till May, and the admin then just have to update them or delete them

**Comment by devsargam** [comment]
I will review again today evening 🎉

**Comment by siinghd** [comment]
@vanshavenger fix lock please.

**Comment by vanshavenger** [comment]
@siinghd the yarn is identical completely, I again copied it from the main and it is showing no difference when i try to commit again for the verification.

**Comment by siinghd** [comment]
@vanshavenger sir look at package-lock and pnpm lock

**Comment by vanshavenger** [comment]
@siinghd donee serr

**Comment by vanshavenger** [comment]
@devsargam @siinghd resolved the merge conflicts.

**Comment by devsargam** [comment]
Checking this

**Comment by devsargam** [comment]
Week tab doesn't work on the calandar
image

**Comment by devsargam** [comment]
Maybe we could just have month + agenda tab. Week and day are just subsets of them.

**Comment by devsargam** [comment]
@hkirat thoughts?

**Comment by vanshavenger** [comment]
Screenshot 2024-08-20 at 1 41 50 PM

@devsargam the week is showing , i just ran the seed script and it is showing the entries.
and da and week come under react-big-calendar only.

**Review by devsargam — APPROVED** [review]

**Comment by vanshavenger** [comment]
@devsargam @hkirat if any further changes are required let me know, happy to resolve them asap :))

File: src/components/big-calendar/calendar.tsx [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+}
+
+const DnDCalendar = withDragAndDrop<CalendarEvent>(ShadcnBigCalendar);
+const localizer = momentLocalizer(moment);
+
+interface LandingPageProps {
+  isAdmin: boolean;
+}
+
+const LandingPage: React.FC<LandingPageProps> = ({ isAdmin }) => {
```
Comment (hkirat) [inline src/components/big-calendar/calendar.tsx:43]: LandingpagE?

File: src/app/calendar/page.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import LandingPage from '@/components/big-calendar/calendar';
```
Comment (hkirat) [inline src/app/calendar/page.tsx:1]: CalendarPage?

Comment (hkirat) [inline src/app/calendar/page.tsx:1]: Move this to /admin/calendar

File: src/actions/event-actions/index.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+export async function fetchEvents() {
+  return await prisma.event.findMany();
+}
+
+export async function addEvent(
+  event: Omit<Event, 'id' | 'createdAt' | 'updatedAt'>,
+) {
+  if (!(await isAdmin())) {
```
Comment (hkirat) [inline src/actions/event-actions/index.ts:23]: good

File: src/components/big-calendar/calendar.tsx (hunk 1 above)
Comment (vanshavenger) [reply src/components/big-calendar/calendar.tsx]: Changed to Calendar Page ser

File: src/app/calendar/page.tsx (hunk 2 above)
Comment (vanshavenger) [reply src/app/calendar/page.tsx]: Changed to calendar Page ser

Comment (vanshavenger) [reply src/app/calendar/page.tsx]: Kirat, in this on the calendar page admin has the option to add event, basically it shows a Add Event button on the calendar, so it is kept this way

File: src/actions/event-actions/index.ts (hunk 3 above)
Comment (vanshavenger) [reply src/actions/event-actions/index.ts:23]: Thanks serrrr

**Pull request merged**